framework/core/allocated.cpp

```diff
@@ -180,6 +180,11 @@ void AllocatedBase::post_create(VmaAllocationInfo const &allocation_info)
 		}
 #endif
 
+	if (VK_IMAGE_COMPRESSION_DEFAULT_EXT != compression_control.flags)
+	{
+		create_info.pNext = &compression_control;
+	}
```

Comment:
so, I feel like this code actually belongs in the `ImageBuilder` class.  i.e. when you create a builder and you call `with_compression`, that's the appropriate time to actually modify the `pNext` chain.

Similarly, I think we should avoid raw `pNext` modification like this.  Compression is unlikely to be the last extension that can be used in `VkImageCreateInfo::pNext` chains that we'll ever do, and correspondingly we should always interact with the chain in a manner that reflects that it might already be populated, and ensure we inject items into the chain, rather than simply overwriting it.

Here, this could be as simple as 
```suggestion
	if (VK_IMAGE_COMPRESSION_DEFAULT_EXT != compression_control.flags)
	{
	        compression_control.pNext = create_info.pNext;
		create_info.pNext = &compression_control;
	}
```

But it's something that should probably be farmed out to a helper function somewhere, because you want to deal with edge cases like both structures having their own chains that need to be merged and runtime checks that a given chain doesn't contain more than one element of the same structure type.

Reply:
Addressed in my last commit